src/datajson.c

```diff
@@ -0,0 +1,1009 @@
+/* Copyright (C) 2025 Open Information Security Foundation
```

Comment:
the json(line) related logic belongs in a `datasets-json(line).c`

Reply:
Is this just `ndjson`? (newline-delimited json)

Reply:
Yes, it is I can reword it in the option.